pkg/controller/controller_instance.go

```diff
@@ -1721,6 +1870,10 @@ func (c *controller) processProvisionFailure(instance *v1beta1.ServiceInstance,
 		errorMessage = fmt.Errorf(readyCond.Message)
 	}
 
+	// assume a provision retry will happen, set a not-before time so we don't pound the Broker
+	// in a constant try to provision/fail/orphan mitigation/repeat loop.
+	c.setNextOperationRetryTime(instance)
```

Comment:
This is only needed for non-terminal errors. For terminal errors you have just removed instance from the retry map in `processTerminalProvisionFailure`.
So the proper place to put this is into the "non-terminal" condition block below: https://github.com/kubernetes-incubator/service-catalog/blob/3837abf27fe5d9b090a7d530a81d6cebc634a79e/pkg/controller/controller_instance.go#L1747-L1752

Reply:
Agreed, thanks